jenkins/post-build-status-update.jenkinsfile

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env groovy
+
+import groovy.json.JsonOutput
+
+pipeline {
+    agent { label 'jenkins-workspace' }
+
+    parameters {
+        string(defaultValue: '', description: 'test/aix, linter, etc.', name: 'IDENTIFIER')
+        string(defaultValue: '', description: 'pending, success, failure', name: 'STATUS')
+        string(defaultValue: '', description: 'URL for upstream Jenkins job', name: 'URL')
+        string(defaultValue: '', description: 'Current commit being tested in upstream Jenkins job', name: 'COMMIT')
+        string(defaultValue: '', description: 'Current branch being tested in upstream Jenkins job', name: 'REF')
+    }
+
+    stages {
+        stage('Send status report') {
+            steps {
+                sendBuildStatus(params.IDENTIFIER, params.STATUS, params.URL, params.COMMIT, params.REF)
+            }
+        }
+    }
+}
+
+def sendBuildStatus(identifier, status, url, commit, ref) {
+    def path = ""
```

Comment:
Not sure exactly where to do it, but it would probably be a good idea to error out if any of the parameters are empty.

Reply:
The 99% case is that this pipeline is triggered by another Jenkins job... is it necessary to have parameter validation? Also, there is no DSL-y way to mark a parameter as required.

Reply:
Yes, but when someone adds a new job they might forget to pass params through, in which case the default `""` will be used.

IDK, I think it'd be helpful.

Can't you just fail the job if they're `""`?